qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,434 @@
+        if self.experiment_options.pauli_randomize:
+            adjusted_2q_density = self.experiment_options.two_qubit_gate_density * 2
+        else:
+            adjusted_2q_density = self.experiment_options.two_qubit_gate_density
+
+        if adjusted_2q_density > 1:
+            warnings.warn("Two-qubit gate density is too high, capping at 1.")
+            adjusted_2q_density = 1
+
+        if self.experiment_options.pauli_randomize:
+            self._distribution.gate_distribution = [
+                GateDistribution(
+                    prob=adjusted_2q_density, op=self.experiment_options.two_qubit_gate
+                ),
+                GateDistribution(prob=1 - adjusted_2q_density, op=GenericClifford(1)),
+            ]
+
+        else:
+            self._distribution.gate_distribution = [
+                GateDistribution(
+                    prob=adjusted_2q_density, op=self.experiment_options.two_qubit_gate
+                ),
+                GateDistribution(prob=1 - adjusted_2q_density, op=GenericClifford(1)),
+            ]
+
+    def _sample_sequences(self) -> List[Sequence[SequenceElementType]]:
+        """Sample layers of mirror RB using the provided distribution and user options.
+
+        First, layers are sampled using the distribution, then Pauli-dressed if
+        ``pauli_randomize`` is ``True``. The inverse of the resulting circuit is
+        appended to the end. If ``start_end_clifford`` is ``True``, then cliffords are added
+        to the beginning and end. If ``inverting_pauli_layer`` is ``True``, a Pauli
+        layer will be appended at the end to set the output bitstring to all zeros.
+
+        Returns:
+            A list of mirror RB sequences. Each element is a list of layers with length
+            matching the corresponding element in ``lengths``. The layers are made up
+            of tuples in the format ((one or more qubit indices), gate). Single-qubit
+            Cliffords are represented by integers for speed.
+
+        Raises:
+            QiskitError: If no backend is provided.
+        """
+        if not self._backend:
+            raise QiskitError("A backend must be provided for circuit generation.")
+
+        self._set_distribution_options()
+
+        # Sequence of lengths to sample for
+        if not self.experiment_options.full_sampling:
+            seqlens = (max(self.experiment_options.lengths),)
+        else:
+            seqlens = self.experiment_options.lengths
+
+        if self.experiment_options.pauli_randomize:
+            pauli_sampler = SingleQubitSampler(seed=self.experiment_options.seed)
+            pauli_sampler.gate_distribution = [GateDistribution(prob=1, op=GenericPauli(1))]
+
+        if self.experiment_options.start_end_clifford:
+            clifford_sampler = SingleQubitSampler(seed=self.experiment_options.seed)
+            clifford_sampler.gate_distribution = [GateDistribution(prob=1, op=GenericClifford(1))]
```

Comment:
Why don't we have `Pauli1QSampler` and `Clifford1QSampler`, which are dedicated samplers for 1Q Paulis and 1Q Cliffords? If we have those, we can simply write `pauli_sampler=Pauli1QSampler(seed=self.experiment_options.seed)` here and readers don't understand the spec of `GateDistribution` and `GenericPauli`.